stp_zmq/zstack.py

```diff
@@ -774,25 +780,31 @@ def send(self, msg: Any, remoteName: str = None, ha=None):
                 return self.transmit(msg, remoteName)
 
     def transmit(self, msg, uid, timeout=None):
-        # Timeout is unused as of now
-        assert uid in self.remotes
-        socket = self.remotes[uid].socket
-        if socket:
-            msg = self.prepMsg(msg)
-            try:
-                # noinspection PyUnresolvedReferences
-                # socket.send(self.signedMsg(msg), flags=zmq.NOBLOCK)
-                socket.send(msg, flags=zmq.NOBLOCK)
-                logger.debug(
-                    '{} transmitting message {} to {}'.format(self, msg, uid))
-                return True
-            except zmq.Again as ex:
-                logger.debug('{} could not transmit message to {}'.format(self, uid))
-                return False
-        else:
-            logger.warning('{} has uninitialised socket for remote {}'.
-                        format(self, self.remotes[uid]))
+        remote = self.remotes.get(uid)
+        if remote is None:
+            logger.error("Remote {} does not exist!".format(uid))
+            return False
+        socket = remote.socket
+        if not socket:
+            logger.info('{} has uninitialised socket '
+                        'for remote {}'.format(uid))
             return False
+        try:
+            msg = self.prepMsg(msg)
+            # socket.send(self.signedMsg(msg), flags=zmq.NOBLOCK)
+            socket.send(msg, flags=zmq.NOBLOCK)
+            logger.info('{} transmitting message {} to {}'
+                        .format(self, msg, uid))
+            if not remote.isConnected:
+                logger.warning('Remote {} is not connected - '
```

Comment:
>Shouldn't we mention that tcp ports must be checked?
I thought about it, but I think that such message is enough, since it is more common

>Shouldn't we move this check before socket.send?
No, because in such case "transmitting message" is going after "can't send", so it looks like message is finally sent.

>Otherwise we will be thrown right into except statement, won't we?
Yes, this is what logic tells, but indeed we won't. This is how zeromq works, unfortunately  

>Do we have a test that checks that this message is logged if we have some problems with tcp port?
No, there is no such test, and I do not see simple way for it (monitoring output or calling of loggers methods?).

This checks not just tcp portt problems, but whole set of connection problems. Zeromq does not provide information about reasons, so we would have to do a number of tests (like ping, trying to open port, resolve address etc) manually if we want such functionality